tools/inc/stljobs.h

```diff
@@ -0,0 +1,606 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <assert.h>
+#include <atomic>
+#include <iterator>
+#include <memory>
+#include <stdexcept>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <string>
+#include <string_view>
+#include <userenv.h>
+
+#include <Windows.h>
+
+struct api_exception : std::exception {
+    const char* api;
+    unsigned long lastError;
+
+    explicit api_exception(const char* api_, unsigned long lastError_) noexcept : api(api_), lastError(lastError_){};
+
+    [[noreturn]] void give_up() const {
+        fflush(stdout);
+        fprintf(stderr, "The API %s failed unexpectedly; last error 0x%08lX\n", api, lastError);
+        abort();
+    }
+
+    const char* what() const noexcept override {
+        return "win32 exception";
+    }
+};
+
+[[noreturn]] inline void api_failure(const char* const api, const unsigned long lastError = GetLastError()) {
+    throw api_exception{api, lastError};
+}
+
+struct invalid_handle_value_policy {
+    static constexpr HANDLE Empty = INVALID_HANDLE_VALUE;
+};
+
+struct null_handle_policy {
+    static constexpr HANDLE Empty = 0;

```

Comment:
It absolutely cannot be `nullptr` because `HANDLE` happening to be a pointer is an implementation detail, and there might even be macro cocktails that make it a struct instead. `{}` is probably OK but doesn't match the rest of the Windows API documentation.

Reply:
I suppose only `-Wzero-as-null-pointer-constant` cares.